### ISSUE TITLE
Very slow on large files

### ISSUE THREAD

**Issue by talex5** [lpw25/prof_alloc#14]
`prof_spacetime process` has now been running on my machine for several hours processing a 5.7G profile. I'm not sure whether to kill it or continue waiting. It would be a shame to give up, as the data took several days to collect.

It would be good if it could be made to run faster, only sample the snapshots, or at least show some progress indicator.

**Comment by talex5** [comment]
(update: I was able to process this file in 45 min by first transferring it to a machine with 64G RAM)

**Comment by lpw25** [comment]
Yeah, I haven't seen it take that long for a profile: it usually runs out of memory instead. I guess you've got a lot of swap space and that is why it was able to continue but taking a very long time.
